internal/distro/rhel8/pipelines.go

```diff
@@ -698,7 +698,17 @@ func osPipeline(t *imageType,
 			},
 		}))
 	}
-
+	var greenbootStageOptions *osbuild.GreenbootConfig
+	if greenbootConfig := c.GetGreenbootConfig(); greenbootConfig != nil {
+		greenbootStageOptions = GreenbootStageOptions(greenbootConfig)
+	}
+	if greenbootStageOptions == nil {
+		defaultGreenbootCustomization := &blueprint.GreenbootCustomization{
+			MonitorServices: []string{"sshd", "NetworkManager"}, //sshd,NetworkManager are default services to be monitored by greenboot
+		}
+		greenbootStageOptions = GreenbootStageOptions(defaultGreenbootCustomization)
+	}
```

Comment:
the logic here is twisted and you're duplicating the default case.. how about:

```go
options := &osbuild.GreenbootConfig{
    MonitorServices: []string{"sshd", "NetworkManager"},
}
if greenbootConfig := c.GetGreenbootConfig(); greenbootConfig != nil {
    if len(config.MonitorServices) != 0 {
        options.MonitorServices = config.MonitorServices
    }
}
```

Reply:
optimised to call the function once.